File: AccuPay/Reports/Reports/Providers/ExcelReportColumn.vb
Public Class ExcelReportColumn

    Public Property Name As String
    Public Property Type As ColumnType
    Public Property Source As String
    Public Property [Optional] As Boolean

    Public Sub New(name As String,
                       source As String,
                       Optional type As ColumnType = ColumnType.Numeric,
                       Optional [optional] As Boolean = False)
        Me.Name = name
        Me.Source = source
        Me.Type = type
        Me.Optional = [optional]
    End Sub

    Public Enum ColumnType
        Text
        Numeric
    End Enum

End Class

File: AccuPay.Infrastructure/Reports/ExcelReportColumn.cs
namespace AccuPay.Infrastructure.Reports
{
    public class ExcelReportColumn
    {
        public string Name { get; set; }
        public ColumnType Type { get; set; }
        public string Source { get; set; }
        public bool Optional { get; set; }

        public ExcelReportColumn(string name, string source, ColumnType type = ColumnType.Numeric, bool optional = false)
        {
            Name = name;
            Source = source;
            Type = type;
            Optional = optional;
        }

        public enum ColumnType
        {
            Text,
            Numeric
        }
    }
}
